bin/aws_cleaner.rb

```diff
@@ -147,14 +152,29 @@ def closelog(message)
       instance_id = AwsCleaner.new.process_message(body)
 
       if instance_id
-        chef_node = AwsCleaner::Chef.get_chef_node_name(instance_id, @config)
-        webhook(id, instance_id)
-        chef(id, instance_id, chef_node)
-        sensu(id, instance_id, chef_node)
+        node = case @config[:lookup_type]
+               when 'sensu'
+                 AwsCleaner::Sensu.get_sensu_node_name(instance_id, @config)
+               else
+                 AwsCleaner::Chef.get_chef_node_name(instance_id, @config)
+               end
+
+        if node
+          @logger.info("Instance #{instance_id} found, node name is #{node}.")
+
+          webhook(instance_id)
+          chef(instance_id, node)
+          sensu(instance_id, node)
+
+          @logger.info("All done with #{node}, deleting message.")
+        else
+          @logger.info("Instance #{instance_id} not found, deleting message.")
+        end
       else
         @logger.info('Message not relevant, deleting')
-        AwsCleaner.new.delete_message(id, @config)
       end
+
+      AwsCleaner.new.delete_message(id, @config)
```

Comment:
I was thinking about this and discussing with a colleague, right now the message is deleted when it is not deemed relevant like when the state is not "terminated", however if you have another process looking at the same SQS queue for "running" events this could cause messages to get lost. Should this be moved to before the else statement so it looks like this? That way the message is only deleted once processed and not when it is ignored, the other side affect is that if you have nothing else listening you will have extra messages queuing up that will never be processed.

```       @logger.info("Instance #{instance_id} not found, deleting message.")
        end

        AwsCleaner.new.delete_message(id, @config)
      else
        @logger.info('Message not relevant, ignoring...')
      end```